src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -74,6 +93,184 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
+                       },
+                   };
+                   await this._store.dispatch(this._orderActionCreator.submitOrder(submitOrderPayload, submitOrderOptions));
+                }
+               await this._paypalCommercePaymentProcessor.deleteCart(cart.id);
+               window.location.assign('/checkout/order-confirmation');
+            } catch (e) {
+                throw new RequestError(e);
+            }
+        }
+
+        return orderPlacement;
+    }
+
+    private async _onShippingChangeHandler(data: ShippingChangeData, actions: ApproveActions, cart: Cart) {
+        const baseOrderAmount = cart.baseAmount;
+        let shippingAmount = '0.00';
+        this._currentShippingAddress = await this._transformToAddress(data.shipping_address);
+        const checkout = this._store.getState().checkout.getCheckoutOrThrow();
+        const availableShippingOptions = checkout.consignments[0].availableShippingOptions;
+        const shippingRequired = checkout.cart.lineItems.physicalItems.length > 0;
+        if (!shippingRequired) {
+            return await actions.order.patch([
+                {
+                    op: 'replace',
+                    path: '/purchase_units/@reference_id==\'default\'/amount',
+                    value: {
+                        currency_code: 'USD',
+                        value: (parseFloat(String(baseOrderAmount))).toFixed(2),
+                        breakdown: {
+                            item_total: {
+                                currency_code: 'USD',
+                                value: baseOrderAmount,
+                            },
+                        },
+                    },
+                },
+            ]);
+        }
+
+        if (shippingRequired && availableShippingOptions?.length === 0) {
+            return actions.reject();
+        }
+
+        const shippingOptions = availableShippingOptions?.map((option: AvaliableShippingOption) => {
+            let isSelected = false;
+            // Buyer has chosen shipping option on PP list and address the same
+            if (this._isAddressSame(this._currentShippingAddress, this._submittedShippingAddress)
+            && option.id === data.selected_shipping_option.id
+            ) {
+            shippingAmount = data.selected_shipping_option.amount.value;
+            isSelected = true;
+            } else if (option.isRecommended) {
+            shippingAmount = parseFloat(String(option.cost)).toFixed(2);
+            isSelected = true;
+            }
+
+            return {
+                id: option.id,
+                type: 'SHIPPING',
+                label: option.description,
+                selected: isSelected,
+                amount: {
+                    value: parseFloat(String(option.cost)).toFixed(2),
+                    currency_code: 'USD',
+                },
+            };
+        });
+        this._submittedShippingAddress = this._currentShippingAddress;
```

Comment:
Add an extra empty line above the 203, please

Reply:
added